tests/regtest.py

```diff
@@ -784,11 +780,26 @@ def getDescription(self):
 		return self.m_test_desc
 
 	def prepare(self):
-		if not self.global_init_performed:
-			# make sure certain dirs exist
-			os.makedirs(self.m_sysconfig, 0o755, exist_ok = True)
-			os.makedirs(self.m_permissions_dir, 0o755, exist_ok = True)
-			self.global_init_performed = True
+		if not TestBase.global_init_performed:
+
+			config_root = self.m_main_test_instance.getChkstatConfigRoot()
+			TestBase.s_config_root = config_root
+			TestBase.s_sysconfig = config_root + "/etc/sysconfig"
+			TestBase.s_sysconfig_security = self.s_sysconfig + "/security"
+			TestBase.s_permissions_dir = config_root + "/etc/permissions.d"
+			TestBase.s_profile_root = config_root + "/usr/share/permissions"
+			TestBase.s_permissions_base = TestBase.s_profile_root + "/permissions"
+			TestBase.s_user_permissions_base = config_root + "/etc/permissions"
+			TestBase.s_chkstat_bin = self.m_main_test_instance.getChkstatPath()
+			# make a convenience symlink to make it feel more
+			# natural in /usr/local
+			os.symlink(config_root + "/usr/share", config_root + "/share")
+
+			# make sure base dirs exist
+			os.makedirs(self.s_sysconfig, 0o755, exist_ok = True)
+			os.makedirs(self.s_permissions_dir, 0o755, exist_ok = True)
+			os.makedirs(self.s_profile_root, 0o755, exist_ok = True)
```

Comment:
Maybe move this into `createTestFile`? That'd already allow getting rid of two of these variables.

Reply:
Hmm I'm not quite sure what you mean yet. `createTestfile` currently is used to create test case specific files somewhere in `/TestCaseDir/testfile`, while these three here create "system configuration directories" that should only be created once and not disappear at any later time.

Reply:
The idea is to do this `mkdir -p` implicitly when the config files are written, so that we can get rid of two of the variables that make it hard to figure out what each path variable refers to. Maybe that's the wrong function, I just saw it being used to create the permissions config files in `resetConfigs`.